feat(perps): add deeplink support for perps

abretonc7s opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR implements deeplink support for the Perps (perpetual futures) feature, enabling direct navigation to Perps markets from external sources like marketing campaigns, notifications, and social media.

The implementation supports two main deeplink types:
1. **Perps Market Overview**: Routes users to the main Perps tab (with tutorial flow for first-time users)
2. **Specific Asset Details**: Routes users directly to a specific perps asset (e.g., BTC, ETH)

## **Changelog**

CHANGELOG entry: Added deeplink support for Perps markets, allowing direct navigation to Perps tab and specific asset details

## **Related issues**

Fixes: TAT-1344

## **Manual testing steps**

### Testing Deeplinks

#### iOS Testing
```bash
# Test Perps market overview deeplink
xcrun simctl openurl booted "https://link-test.metamask.io/perps"

# Test specific asset deeplinks
xcrun simctl openurl booted "https://link-test.metamask.io/perps-asset?symbol=BTC"
xcrun simctl openurl booted "https://link-test.metamask.io/perps-asset?symbol=ETH"
xcrun simctl openurl booted "https://link-test.metamask.io/perps-asset?symbol=SOL"
```

#### Android Testing
```bash
# Test Perps market overview deeplink
adb shell am start -W -a android.intent.action.VIEW -d "https://link-test.metamask.io/perps" io.metamask.debug

# Test specific asset deeplinks
adb shell am start -W -a android.intent.action.VIEW -d "https://link-test.metamask.io/perps-asset?symbol=BTC" io.metamask.debug
adb shell am start -W -a android.intent.action.VIEW -d "https://link-test.metamask.io/perps-asset?symbol=ETH" io.metamask.debug
```

https://github.com/user-attachments/assets/d391fca1-be2a-4d65-96b2-2c5f2ef2ae34


### Resetting First-Time User State

To test the tutorial flow for first-time users, you need to reset the Perps state:

1. **Via Redux DevTools (Development builds):**
   - Open the app with Redux DevTools enabled
   - Navigate to Redux state
   - Find `engine.backgroundState.PerpsController`
   - Set `isFirstTimeUser: { testnet: true, mainnet: true }`

2. **Via Settings Menu:**
   - Go to Settings → Advanced → Reset Account
   - This will reset all account data including Perps state


### Test Scenarios

```gherkin
Feature: Perps Deeplinks Navigation

  Scenario: First-time user navigates via perps deeplink
    Given the user has MetaMask Mobile installed
    And the user has never used Perps before (reset state if needed)

    When user clicks on deeplink "https://link-test.metamask.io/perps"
    Then the app opens to the Perps Tutorial screen
    And after completing or skipping the tutorial, user sees the Perps tab selected

  Scenario: Returning user navigates via perps deeplink
    Given the user has MetaMask Mobile installed
    And the user has completed the Perps tutorial

    When user clicks on deeplink "https://link-test.metamask.io/perps"
    Then the app opens directly to the Wallet home with Perps tab selected

  Scenario: User navigates to specific BTC asset via deeplink
    Given the user has MetaMask Mobile installed

    When user clicks on deeplink "https://link-test.metamask.io/perps-asset?symbol=BTC"
    Then the app opens directly to the BTC perps market details screen

  Scenario: User navigates to specific ETH asset via deeplink
    Given the user has MetaMask Mobile installed

    When user clicks on deeplink "https://link-test.metamask.io/perps-asset?symbol=ETH"
    Then the app opens directly to the ETH perps market details screen

  Scenario: User navigates with invalid asset symbol
    Given the user has MetaMask Mobile installed

    When user clicks on deeplink "https://link-test.metamask.io/perps-asset?symbol=INVALID"
    Then the app opens to the Perps tab (fallback behavior)

  Scenario: Tutorial skip navigates correctly from deeplink
    Given the user is a first-time Perps user
    And the user arrived via deeplink

    When user skips the tutorial
    Then the app navigates to Wallet home with Perps tab selected
    And the user can see the Perps markets list
```

### Expected Results

1. **First-time users**: Should see the tutorial carousel with 6 steps, then navigate to Perps tab
2. **Returning users**: Should navigate directly to Wallet home with Perps tab selected
3. **Asset deeplinks**: Should open the specific market details view
4. **Invalid symbols**: Should fallback to Perps markets list
5. **Tab selection**: The Perps tab should be visually selected and active after navigation

## **Screenshots/Recordings**

### **Before**
- No deeplink support for Perps feature
- Users had to manually navigate to Perps tab through the app

### **After**
- Direct deeplink navigation to Perps markets
- Support for specific asset deeplinks (BTC, ETH, SOL, etc.)
- Smart routing based on user state (tutorial vs direct navigation)

https://github.com/user-attachments/assets/28b55198-995a-4e55-abc1-6715e528d9ac


## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

abretonc7s committed:
feat: handle peprs deeplinks

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/deeplinks

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/deeplinks

abretonc7s committed:
feat: handle navigation to perps tab

abretonc7s committed:
chore: cleanup

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/deeplinks

abretonc7s committed:
feat: coverage

abretonc7s committed:
fix: ts errors

abretonc7s committed:
fix: unit tests

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/deeplinks

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/deeplinks

abretonc7s committed:
fix: unit tests

abretonc7s committed:
feat: cleanup

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/deeplinks

Cal-L commented on PR_DESC.md:
Is this PR doc file a new pattern or is the plan to remove this before this is merged?

Cal-L commented on app/components/UI/Perps/components/PerpsTutorialCarousel/PerpsTutorialCarousel.tsx:
```diff
@@ -203,8 +218,31 @@ const PerpsTutorialCarousel: React.FC = () => {
+
+      // The timeout is REQUIRED - React Navigation needs time to complete
+      // the navigation before params can be set on the new screen
+      setTimeout(() => {
```
We should be able to provide params to the Wallet route instead of setting params separately, eliminating the need for the set timeout below.

Cal-L reviewed (commented):
Can you share some videos of the deeplinks in action?

abretonc7s committed:
feat: add Reset Account support for Perps first-time user state

- Add resetFirstTimeUserState method to PerpsController to reset tutorial flags
- Integrate with Reset Account feature in settings for easier testing
- Add comprehensive unit tests for the reset functionality
- Remove PR_DESC.md from repository (keep locally for reference)

abretonc7s committed:
refactor: use hook for resetFirstTimeUserState instead of Engine.context

Improves separation of concerns by using the usePerpsFirstTimeUser hook
in ResetAccountModal instead of directly accessing Engine.context

abretonc7s committed:
test: add unit tests for resetFirstTimeUserState in usePerpsFirstTimeUser hook

Added comprehensive tests for the new resetFirstTimeUserState function
Ensures proper mocking and edge case handling for undefined controller

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/deeplinks

abretonc7s committed:
test: add coverage for Perps deeplink handling

Added test coverage for newly introduced Perps deeplink functionality:
- CREATE_ACCOUNT action handler tests
- PERPS and PERPS_MARKETS action handler tests
- PERPS_ASSET action handler tests
- DeeplinkManager methods for _handlePerps and _handlePerpsAsset

abretonc7s committed:
test: add minimal coverage for Perps deeplink navigation

Added test coverage for Perps-specific code added in this PR:
- Perps and PERPS_MARKETS deeplink handling
- PERPS_ASSET deeplink handling
- Perps tab selection via navigation params in Wallet component
- useFocusEffect handling for shouldSelectPerpsTab param

abretonc7s committed:
fix: coverage

abretonc7s committed:
feat: coverage

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/deeplinks

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/deeplinks

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/deeplinks

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/deeplinks

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/deeplinks

abretonc7s committed:
feat: coverage

abretonc7s committed:
fix: unit tests

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/deeplinks

abretonc7s committed:
merge latest main

abretonc7s committed:
fix: test from merge

abretonc7s committed:
feat: cleanup

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/deeplinks

abretonc7s commented:
> I've found a couple of things:
> 
> * Android is not working for me. Not locally or when I built a release build. How did you get it working?
> * on iOS when I navigate to BTC-USD, the screen shifts when it finishes loading. Also when I go to add funds, the header navigation changes from having a back button to just add funds
> * Can I clarify if https://link-test.metamask.io/perps-asset?symbol=BTC will take first time users to tutorial pages as well? I ran into a situation where https://link-test.metamask.io/perps is taking me to "what are perps" screen and https://link-test.metamask.io/perps-asset?symbol=BTC is taking to me BTC-USD, is this expected?
> 
> ![Android Perps](https://private-user-images.githubusercontent.com/212610716/481787199-8bdf09e0-3d49-4490-b962-ea3ce8f5be76.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYxNjkwMzgsIm5iZiI6MTc1NjE2ODczOCwicGF0aCI6Ii8yMTI2MTA3MTYvNDgxNzg3MTk5LThiZGYwOWUwLTNkNDktNDQ5MC1iOTYyLWVhM2NlOGY1YmU3Ni5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyNlQwMDM4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYWNlN2JiNDE1YmZhNjhkYTJiNWUzMGFlMTU0MDA5YTBjZjI1M2QzMzhkM2RhMjc5Mzg0NWUyMWZkOTU2MzM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.bEVrKXv4Lg1dI3gOK4rgufFfE9Vvu70DVweLvM9jV4U) [ ![Android Perps](https://private-user-images.githubusercontent.com/212610716/481787199-8bdf09e0-3d49-4490-b962-ea3ce8f5be76.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYxNjkwMzgsIm5iZiI6MTc1NjE2ODczOCwicGF0aCI6Ii8yMTI2MTA3MTYvNDgxNzg3MTk5LThiZGYwOWUwLTNkNDktNDQ5MC1iOTYyLWVhM2NlOGY1YmU3Ni5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyNlQwMDM4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYWNlN2JiNDE1YmZhNjhkYTJiNWUzMGFlMTU0MDA5YTBjZjI1M2QzMzhkM2RhMjc5Mzg0NWUyMWZkOTU2MzM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.bEVrKXv4Lg1dI3gOK4rgufFfE9Vvu70DVweLvM9jV4U) ](https://private-user-images.githubusercontent.com/212610716/481787199-8bdf09e0-3d49-4490-b962-ea3ce8f5be76.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYxNjkwMzgsIm5iZiI6MTc1NjE2ODczOCwicGF0aCI6Ii8yMTI2MTA3MTYvNDgxNzg3MTk5LThiZGYwOWUwLTNkNDktNDQ5MC1iOTYyLWVhM2NlOGY1YmU3Ni5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyNlQwMDM4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYWNlN2JiNDE1YmZhNjhkYTJiNWUzMGFlMTU0MDA5YTBjZjI1M2QzMzhkM2RhMjc5Mzg0NWUyMWZkOTU2MzM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.bEVrKXv4Lg1dI3gOK4rgufFfE9Vvu70DVweLvM9jV4U) [ ](https://private-user-images.githubusercontent.com/212610716/481787199-8bdf09e0-3d49-4490-b962-ea3ce8f5be76.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYxNjkwMzgsIm5iZiI6MTc1NjE2ODczOCwicGF0aCI6Ii8yMTI2MTA3MTYvNDgxNzg3MTk5LThiZGYwOWUwLTNkNDktNDQ5MC1iOTYyLWVhM2NlOGY1YmU3Ni5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyNlQwMDM4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYWNlN2JiNDE1YmZhNjhkYTJiNWUzMGFlMTU0MDA5YTBjZjI1M2QzMzhkM2RhMjc5Mzg0NWUyMWZkOTU2MzM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.bEVrKXv4Lg1dI3gOK4rgufFfE9Vvu70DVweLvM9jV4U) ![BTC Perps](https://private-user-images.githubusercontent.com/212610716/481787243-99bead52-8949-436b-b2e9-7e844dcf1ebd.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYxNjkwMzgsIm5iZiI6MTc1NjE2ODczOCwicGF0aCI6Ii8yMTI2MTA3MTYvNDgxNzg3MjQzLTk5YmVhZDUyLTg5NDktNDM2Yi1iMmU5LTdlODQ0ZGNmMWViZC5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyNlQwMDM4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MzQ5MzgyZmQwMWJhYzNiZWNjZmI3OTllY2ZlZjA2ZTJhMmI2YTdiZjhiNjE1OThjNjVjNGRkNTlmYmEwNTQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.5sk0EKxqzfMSujj2Gm9elKD5yZPUSVuSrvnj0W7u_84) [ ![BTC Perps](https://private-user-images.githubusercontent.com/212610716/481787243-99bead52-8949-436b-b2e9-7e844dcf1ebd.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYxNjkwMzgsIm5iZiI6MTc1NjE2ODczOCwicGF0aCI6Ii8yMTI2MTA3MTYvNDgxNzg3MjQzLTk5YmVhZDUyLTg5NDktNDM2Yi1iMmU5LTdlODQ0ZGNmMWViZC5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyNlQwMDM4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MzQ5MzgyZmQwMWJhYzNiZWNjZmI3OTllY2ZlZjA2ZTJhMmI2YTdiZjhiNjE1OThjNjVjNGRkNTlmYmEwNTQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.5sk0EKxqzfMSujj2Gm9elKD5yZPUSVuSrvnj0W7u_84) ](https://private-user-images.githubusercontent.com/212610716/481787243-99bead52-8949-436b-b2e9-7e844dcf1ebd.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYxNjkwMzgsIm5iZiI6MTc1NjE2ODczOCwicGF0aCI6Ii8yMTI2MTA3MTYvNDgxNzg3MjQzLTk5YmVhZDUyLTg5NDktNDM2Yi1iMmU5LTdlODQ0ZGNmMWViZC5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyNlQwMDM4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MzQ5MzgyZmQwMWJhYzNiZWNjZmI3OTllY2ZlZjA2ZTJhMmI2YTdiZjhiNjE1OThjNjVjNGRkNTlmYmEwNTQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.5sk0EKxqzfMSujj2Gm9elKD5yZPUSVuSrvnj0W7u_84) [ ](https://private-user-images.githubusercontent.com/212610716/481787243-99bead52-8949-436b-b2e9-7e844dcf1ebd.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYxNjkwMzgsIm5iZiI6MTc1NjE2ODczOCwicGF0aCI6Ii8yMTI2MTA3MTYvNDgxNzg3MjQzLTk5YmVhZDUyLTg5NDktNDM2Yi1iMmU5LTdlODQ0ZGNmMWViZC5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyNlQwMDM4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MzQ5MzgyZmQwMWJhYzNiZWNjZmI3OTllY2ZlZjA2ZTJhMmI2YTdiZjhiNjE1OThjNjVjNGRkNTlmYmEwNTQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.5sk0EKxqzfMSujj2Gm9elKD5yZPUSVuSrvnj0W7u_84)

On android you need to enable the links for metamask on dev builds (it only works directly with the signed apk)

abretonc7s committed:
feat: merge main into branch

gantunesr commented on app/components/Views/Settings/AdvancedSettings/ResetAccountModal/ResetAccountModal.tsx:
```diff
@@ -28,13 +29,16 @@ export const ResetAccountModal = ({
       wipeSmartTransactions(selectedAddress);
     }
     wipeTransactions();
+    // Reset Perps first-time user state for testing
+    resetFirstTimeUserState();
```
Can this be added to the unit test from this file?

abretonc7s committed:
Merge branch 'main' into perps/feat/deeplinks

abretonc7s committed:
Merge branch 'main' into perps/feat/deeplinks

nickewansmith commented:
@abretonc7s Let's make sure to circle back on anything mentioned that is not addressed in the PR.
